src/gui/handler.hpp

```diff
@@ -0,0 +1,20 @@
+// handles fbo, and other things if necessary
+
+#include <GL/glew.h>
+
+namespace gui {
+
+class FrameBufferHandler {
+ private:
+    static GLuint fbo_name;
+
+ public:
+    static inline GLuint
+    get_current_fbo() {
+        return fbo_name;
+    }
+
+    static void bind_fbo(GLuint new_fbo);
+};
```

Comment:
You're doing this wrong. Instead of having all data static, you want this to be a singleton. 
```cpp
class FrameBufferHandler {
 private:
    GLuint fbo_id;

    // Private CTOR as this is a singleton
    FrameBufferHandler() : fbo_id(0) {}

 public:
    // Delete all CTORs and CTOR-like operators
    FrameBufferHandler(FrameBufferHandler&&)        = delete;
    FrameBufferHandler(FrameBufferHandler const&)   = delete;

    void operator=(FrameBufferHandler&&)            = delete;
    void operator=(FrameBufferHandler const&)       = delete;

    // Instance accessor
    static inline
    FrameBufferHandler& instance() {
        static FrameBufferHandler obj;
        return obj;
    }

    inline GLuint
    get_current_fbo() {
        return fbo_id;
    }

    void inline
    bind_fbo(GLuint new_fbo_id) {
        if (new_fbo_id == fbo_id) {
            // already bound
            return;
        } else {
            fbo_id = new_fbo_id;
            glBindFramebuffer(GL_FRAMEBUFFER, new_fbo_id);
        }
    }
};
```